client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/MlClientDocumentationIT.java

```diff
@@ -2802,6 +2786,298 @@ public void onFailure(Exception e) {
+            // end::get-data-frame-analytics-stats-request
+
+            // tag::get-data-frame-analytics-stats-execute
+            GetDataFrameAnalyticsStatsResponse response =
+                client.machineLearning().getDataFrameAnalyticsStats(request, RequestOptions.DEFAULT);
+            // end::get-data-frame-analytics-stats-execute
+
+            // tag::get-data-frame-analytics-stats-response
+            List<DataFrameAnalyticsStats> stats = response.getAnalyticsStats();
+            // end::get-data-frame-analytics-stats-response
+
+            assertThat(stats.size(), equalTo(1));
+        }
+        {
+            GetDataFrameAnalyticsStatsRequest request = new GetDataFrameAnalyticsStatsRequest("my-analytics-config");
+
+            // tag::get-data-frame-analytics-stats-execute-listener
+            ActionListener<GetDataFrameAnalyticsStatsResponse> listener = new ActionListener<>() {
+                @Override
+                public void onResponse(GetDataFrameAnalyticsStatsResponse response) {
+                    // <1>
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    // <2>
+                }
+            };
+            // end::get-data-frame-analytics-stats-execute-listener
+
+            // Replace the empty listener by a blocking listener in test
+            CountDownLatch latch = new CountDownLatch(1);
+            listener = new LatchedActionListener<>(listener, latch);
+
+            // tag::get-data-frame-analytics-stats-execute-async
+            client.machineLearning().getDataFrameAnalyticsStatsAsync(request, RequestOptions.DEFAULT, listener); // <1>
+            // end::get-data-frame-analytics-stats-execute-async
+
+            assertTrue(latch.await(30L, TimeUnit.SECONDS));
+        }
+    }
+
+    private static final DataFrameAnalyticsConfig DF_ANALYTICS_CONFIG =
+        DataFrameAnalyticsConfig.builder("my-analytics-config")
+            .setSource(DataFrameAnalyticsSource.builder()
+                .setIndex("put-test-source-index")
+                .build())
+            .setDest(DataFrameAnalyticsDest.builder()
+                .setIndex("put-test-dest-index")
+                .build())
+            .setAnalysis(OutlierDetection.createDefault())
+            .build();
+
+    public void testPutDataFrameAnalytics() throws Exception {
+        RestHighLevelClient client = highLevelClient();
+        {
+            // tag::put-data-frame-analytics-query-config
+            QueryConfig queryConfig = new QueryConfig(new MatchAllQueryBuilder());
+            // end::put-data-frame-analytics-query-config
+
+            // tag::put-data-frame-analytics-source-config
+            DataFrameAnalyticsSource sourceConfig = DataFrameAnalyticsSource.builder()
+                .setIndex("put-test-source-index")
+                .setQueryConfig(queryConfig)
```

Comment:
I think it would be a good idea to have callouts here and in the docs, and use those callouts to reference default values, format, etc.